services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,43 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  // If no event is given arguments are bad
+  if (event == NULL) return ERR_CODE_INVALID_ARG;
+  // If queue handle doesnt describe a queue, error in state
+  if (thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_STATE;
+  // If queue is full, cannot add event, error
+  if (uxQueueSpacesAvailable(thermalMgrQueueHandle) == 0) return ERR_CODE_QUEUE_FULL;
+  xQueueSend(thermalMgrQueueHandle, event, 10);
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  error_code_t errCode;
+  float tempReading = 0.0;
+  //Imported the logging functions to use this function
+  RETURN_IF_ERROR_CODE(readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempReading));
+  if (tempReading>=80)
+    overTemperatureDetected();
+  else if (tempReading<=75)
+    safeOperatingConditions();
 }
```

Comment:
This handler is executed in an ISR so you shouldn't call readTempLM75BD. readTempLM75BD involves I2C read/writes which are slow and ISRs shouldn't take too long to execute.

Another way to do this would be to send an event to the thermal mgr and the mgr will handle it outside the ISR.